File: csharp/MapleStory.OpenAPI/Src/Dto/Notice/CashshopNoticeListItemDTO.cs
using Newtonsoft.Json;
using System.Text.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐시샵 공지
    /// </summary>
    public class CashshopNoticeListItemDTO
    {
        /// <summary>
        /// 공지 제목
        /// </summary>
        [JsonProperty("title")]
        public string Title { get; set; }

        /// <summary>
        /// 공지 링크
        /// </summary>
        [JsonProperty("url")]
        public string Url { get; set; }

        /// <summary>
        /// 공지 식별자
        /// </summary>
        [JsonProperty("notice_id")]
        public int NoticeId { get; set; }

        /// <summary>
        /// 공지 등록일
        /// </summary>
        [JsonProperty("date")]
        public DateTimeOffset Date
        {
            get
            {
                return _date.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _date = value;
            }
        }

        private DateTimeOffset _date;

        /// <summary>
        /// 판매 시작일
        /// </summary>
        [JsonProperty("date_sale_start")]
        public DateTimeOffset? DateSaleStart
        {
            get
            {
                return _dateSaleStart?.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _dateSaleStart = value;
            }
        }

        private DateTimeOffset? _dateSaleStart;

        /// <summary>
        /// 판매 종료일
        /// </summary>
        [JsonProperty("date_sale_end")]
        public DateTimeOffset? DateSaleEnd
        {
            get
            {
                return _dateSaleEnd?.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _dateSaleEnd = value;
            }
        }

        private DateTimeOffset? _dateSaleEnd;

        /// <summary>
        /// 상시 판매 여부 (true - 상시)
        /// </summary>
        [JsonProperty("ongoing_flag")]
        public string OnGoingFlag { get; set; }

        /// <summary>
        /// 상시 판매 여부 (true - 상시)
        /// </summary>
        public bool IsOnGoing
        {
            get
            {
                return OnGoingFlag == "true";
            }
        }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/kms/dto/notice/CashshopNoticeListItemDTO.java
package dev.spiralmoon.maplestory.api.kms.dto.notice;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;

/**
 * 캐시샵 공지
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CashshopNoticeListItemDTO {

    /**
     * 공지 제목
     */
    @JsonProperty("title")
    private String title;

    /**
     * 공지 링크
     */
    @JsonProperty("url")
    private String url;

    /**
     * 썸네일 링크
     */
    @JsonProperty("thumbnail_url")
    private String thumbnailUrl;

    /**
     * 공지 식별자
     */
    @JsonProperty("notice_id")
    private int noticeId;

    /**
     * 공지 등록일
     */
    @JsonProperty("date")
    private String date;

    /**
     * 판매 시작일
     */
    @JsonProperty("date_sale_start")
    private String dateSaleStart;

    /**
     * 판매 종료일
     */
    @JsonProperty("date_sale_end")
    private String dateSaleEnd;

    /**
     * 상시 판매 여부 (true - 상시)
     */
    @JsonProperty("ongoing_flag")
    private String ongoingFlag;

    /**
     * 공지 등록일
     */
    public ZonedDateTime getDate() {
        return Utils.toZonedDateTime(this.date);
    }

    /**
     * 판매 시작일
     */
    public ZonedDateTime getDateSaleStart() {
        return this.dateSaleStart != null ? Utils.toZonedDateTime(this.dateSaleStart) : null;
    }

    /**
     * 판매 종료일
     */
    public ZonedDateTime getDateSaleEnd() {
        return this.dateSaleEnd != null ? Utils.toZonedDateTime(this.dateSaleEnd) : null;
    }

    /**
     * 상시 판매 여부 (true - 상시)
     */
    public boolean isOnGoing() {
        return "true".equals(this.ongoingFlag);
    }
}
